### ISSUE TITLE
Using spacy.TorchBiLSTMEncoder.v1 in tok2vec encode throws error when used for Pretraining

### ISSUE THREAD

**Issue by ksingh196** [explosion/spaCy#7088]
## How to reproduce the behaviour

For the encode method in tok2vec I tried using spacy.TorchBiLSTMEncoder.v1 . After that I tried pretraining but I am getting the below mentioned error :
`ValueError: Attempt to change dimension 'nI' for model 'maxout' from 64 to 128`

I am adding my config file below:
```
[paths]
train = null
dev = nul
vectors = null
init_tok2vec = null
raw_text = "path/to/jsonl/data"

[system]
gpu_allocator = "pytorch"
seed = 0

[nlp]
lang = "en"
pipeline = ["tok2vec","ner"]
batch_size = 1000
disabled = []
before_creation = null
after_creation = null
after_pipeline_creation = null
tokenizer = {"@tokenizers":"spacy.Tokenizer.v1"}

[components]

[components.ner]
factory = "ner"
moves = null
update_with_oracle_cut_size = 100

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
extra_state_tokens = false
hidden_width = 128
maxout_pieces = 2
use_upper = true
nO = null

[components.ner.model.tok2vec]
@architectures = "spacy.Tok2VecListener.v1"
width = 128
upstream = "*"

[components.tok2vec]
factory = "tok2vec"

[components.tok2vec.model]
@architectures = "spacy.Tok2Vec.v2"

[components.tok2vec.model.embed]
@architectures = "spacy.MultiHashEmbed.v1"
width = 128
attrs = ["NORM","PREFIX","SUFFIX","SHAPE"]
rows = [10000,5000,5000,5000]
include_static_vectors = true

[components.tok2vec.model.encode]
@architectures = "spacy.TorchBiLSTMEncoder.v1"
width = 128
depth = 1
dropout = 0.2

[corpora]

[corpora.dev]
@readers = "spacy.Corpus.v1"
path = ${paths.dev}
max_length = 0
gold_preproc = false
limit = 0
augmenter = null

[corpora.pretrain]
@readers = "spacy.JsonlCorpus.v1"
path = ${paths.raw_text}
min_length = 5
max_length = 500
limit = 0

[corpora.train]
@readers = "spacy.Corpus.v1"
path = ${paths.train}
max_length = 2000
gold_preproc = false
limit = 0
augmenter = null

[training]
dev_corpus = "corpora.dev"
train_corpus = "corpora.train"
seed = ${system.seed}
gpu_allocator = ${system.gpu_allocator}
dropout = 0.1
accumulate_gradient = 1
patience = 2000
max_epochs = 0
max_steps = 40000
eval_frequency = 200
frozen_components = []
before_to_disk = null

[training.batcher]
@batchers = "spacy.batch_by_words.v1"
discard_oversize = false
tolerance = 0.2
get_length = null

[training.batcher.size]
@schedules = "compounding.v1"
start = 100
stop = 1000
compound = 1.001
t = 0.0

[training.logger]
@loggers = "spacy.ConsoleLogger.v1"
progress_bar = false

[training.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = false
eps = 0.00000001
learn_rate = 0.001

[training.score_weights]
ents_per_type = null
ents_f = 1.0
ents_p = 0.0
ents_r = 0.0

[pretraining]
max_epochs = 350
dropout = 0.2
n_save_every = null
component = "tok2vec"
layer = ""
corpus = "corpora.pretrain"

[pretraining.batcher]
@batchers = "spacy.batch_by_words.v1"
size = 10000
discard_oversize = false
tolerance = 0.2
get_length = null

[pretraining.objective]
@architectures = "spacy.PretrainVectors.v1"
maxout_pieces = 3
hidden_size = 300
loss = "cosine"

[pretraining.optimizer]
@optimizers = "Adam.v1"
beta1 = 0.9
beta2 = 0.999
L2_is_weight_decay = true
L2 = 0.01
grad_clip = 1.0
use_averages = true
eps = 0.00000001
learn_rate = 0.001

[initialize]
vectors = "en_core_web_lg"
init_tok2vec = ${paths.init_tok2vec}
vocab_data = null
lookups = null
before_init = null
after_init = null

[initialize.components]

[initialize.tokenizer]
```

## Your Environment

* Python Version Used: 3.8
* spaCy Version Used: v3

**Comment by svlandeg** [comment]
Which exact versions of spaCy and Thinc do you have installed?

**Comment by ksingh196** [comment]
spacy '3.0.1'
thinc  '8.0.1'

**Comment by svlandeg** [comment]
Ok, could you try upgrading `spacy` to 3.0.3? Because that should have the fix for a previous bug report https://github.com/explosion/spaCy/issues/6946 which feels very similar.

**Comment by ksingh196** [comment]
same error
ValueError: Attempt to change dimension 'nI' for model 'maxout' from 64 to 128

**Comment by svlandeg** [comment]
Ah, shoot. I'll look into this, could be a related bug then.

**Comment by svlandeg** [comment]
Hi again! Sorry this took a little longer than anticipated to debug.

I think this may be due to an inconsistency in Thinc - we're looking into it, and I'll let you know once this gets fixed.

**Comment by ksingh196** [comment]
Thanks. :)

**Comment by svlandeg** [comment]
This will be fixed in spacy 3.0.4!
